File: src/main/java/com/company/erp/core/response/ErrorCodes.java
package com.company.erp.core.response;

/** Stable machine-readable error codes returned in {@link ApiResponse#errorCode()}. */
public final class ErrorCodes {

    private ErrorCodes() {}

    public static final String VALIDATION_FAILED  = "VALIDATION_FAILED";
    public static final String UNAUTHORIZED       = "UNAUTHORIZED";
    public static final String FORBIDDEN          = "FORBIDDEN";
    public static final String NOT_FOUND          = "NOT_FOUND";
    public static final String CONFLICT           = "CONFLICT";
    public static final String BAD_REQUEST        = "BAD_REQUEST";
    public static final String RATE_LIMITED       = "RATE_LIMITED";
    public static final String UNSUPPORTED_MEDIA  = "UNSUPPORTED_MEDIA_TYPE";
    public static final String METHOD_NOT_ALLOWED = "METHOD_NOT_ALLOWED";
    public static final String INTERNAL_ERROR     = "INTERNAL_ERROR";
}

File: app/Support/Response/ErrorCodes.php
<?php

namespace App\Support\Response;

/** Stable machine-readable error codes returned in the envelope's `errorCode`. */
final class ErrorCodes
{
    const VALIDATION_FAILED  = 'VALIDATION_FAILED';
    const UNAUTHORIZED       = 'UNAUTHORIZED';
    const FORBIDDEN          = 'FORBIDDEN';
    const NOT_FOUND          = 'NOT_FOUND';
    const CONFLICT           = 'CONFLICT';
    const BAD_REQUEST        = 'BAD_REQUEST';
    const RATE_LIMITED       = 'RATE_LIMITED';
    const UNSUPPORTED_MEDIA  = 'UNSUPPORTED_MEDIA_TYPE';
    const METHOD_NOT_ALLOWED = 'METHOD_NOT_ALLOWED';
    const INTERNAL_ERROR     = 'INTERNAL_ERROR';

    private function __construct()
    {
    }
}
